game-core/src/main/java/games/strategy/triplea/delegate/battle/steps/fire/FireRoundStepsBuilder.java

```diff
@@ -0,0 +1,71 @@
+package games.strategy.triplea.delegate.battle.steps.fire;
+
+import games.strategy.engine.delegate.IDelegateBridge;
+import games.strategy.triplea.delegate.DiceRoll;
+import games.strategy.triplea.delegate.battle.BattleActions;
+import games.strategy.triplea.delegate.battle.BattleState;
+import games.strategy.triplea.delegate.battle.MustFightBattle;
+import games.strategy.triplea.delegate.battle.steps.BattleStep;
+import games.strategy.triplea.delegate.data.CasualtyDetails;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+import lombok.Builder;
+import lombok.NonNull;
+import lombok.experimental.UtilityClass;
+
+/** Build the steps for the fire round (roll dice, select casualties, and mark casualties) */
+@UtilityClass
+public class FireRoundStepsBuilder {
+
+  @Builder
+  public static class Parameters {
+    @NonNull final BattleState battleState;
+    @NonNull final BattleActions battleActions;
+    @NonNull final Function<BattleState, List<FiringGroup>> firingGroupSplitter;
+    @NonNull final BattleState.Side side;
+    @NonNull final MustFightBattle.ReturnFire returnFire;
+    @NonNull final BiFunction<IDelegateBridge, RollDice, DiceRoll> roll;
+    @NonNull final BiFunction<IDelegateBridge, SelectCasualties, CasualtyDetails> selectCasualties;
+  }
+
+  public static List<BattleStep> buildSteps(final Parameters parameters) {
```

Comment:
I think you could fit the builder pattern more exactly by removing the parameter object, adding `@Builder` to `FireRoundStepsBuilder`, and then convert this 'buildSteps' method to be a no-arg.

Even doing that, I think this class would still be more of a factory. The following flow is a bit odd:

```
*Builder.builder()
     .
     .
     .build()
     .buildSteps(...)
```

If we call this a factory though, I think it works out better and is arguably a better fit to the pattern (a builder tends to return the thing that is prefixed on the builder, EG: a stringBuilder returns a string, a "gameBuilder' returns a 'game')

Calling this a factory, we get something like:

```
FireRoundStepsFactory.builder()
             .
             .
             .build()
             .createSteps();
```